source/Calamari/Deployment/PackageRetention/Model/Journal.cs

```diff
@@ -123,9 +150,28 @@ public int GetUsageCount(PackageIdentity package)
             return GetUsage(package).Count();
         }
 
-        public void ExpireStaleLocks()
+        public void ExpireStaleLocks(TimeSpan timeBeforeExpiration)
         {
-            throw new NotImplementedException();
+            try
+            {
+                using (var repository = repositoryFactory.CreateJournalRepository())
+                {
+                    foreach (var entry in repository.GetAllJournalEntries())
+                    {
+                        var usages = entry.GetLockDetails();
```

Comment:
Should probably rename `usages` to `locks` now.  (It's a touch nit-picky, but I'm just matching my pedantry with your improving skill - the better you get, the more I'm going to try to get you to improve...)